src/openfermion/hamiltonians/_general_hubbard.py

```diff
@@ -39,6 +39,8 @@ def tunneling_operator(i, j, coefficient):
     return (FermionOperator(((i, 1), (j, 0)), coefficient) + 
             FermionOperator(((j, 1), (i, 0)), coefficient.conjugate()))
 
+def magnetic_field_operator(i, coefficient=1.):
+    return number_operator(i, coefficient) - number_operator(i + 1, coefficient)
```

Comment:
Isn't this assuming that i is up and i+1 is down? That seems incompatible with the more general indexing schemes you're supporting.

Reply:
Ok, but there's a good chance we'll ultimately change the ordering to `(spin_index, site_index, dof_index)`. This ordering results in lower depth (by a factor of 2) circuits for Bogoliubov transformations in which orbitals of different spin don't interact. Most of your code is independent of the ordering, right? So I think we might as well make this one independent too.